automation/get_target_servers.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/local/bin/python
+"""Script to get target server from Apigee"""
+
+import argparse
+import zipfile
+import io
+from requests import Response
+from requests import Session
+
+from service import apigee_auth
+
+# Global session used for all requests.
+REQUEST = Session()
+APIGEE_API_URL = 'https://api.enterprise.apigee.com/v1/organizations/{}/environments/{}'
+
+
+class TargetServer:
+    def __init__(self, organization: str, environment: str, target_server: any):
+        self.organization = organization
+        self.environment = environment
+        self.target_server = target_server
+
+    def __str__(self):
+        return f"TargetServer: organization: {self.organization}, environment: {self.environment}, target_server: {self.target_server}"
+
+
+def get_target_server(org_name: str, env_name: str, target_server_name: str):
+    """Retrieves the target server by name of the given organization name and environment name"""
+
+    response = REQUEST.get((APIGEE_API_URL + '/targetservers/{}').format(org_name, env_name, target_server_name))
+
+    if response.status_code != 200:
+        raise Exception(print_error(response))
```

Comment:
Wouldn't a `404` be an acceptable response as well? If the search is being done across multiple organizations and environments, we want to be able to ignore the `Not Found` error, since it is just saying that it isn't present for that specific combination.

The result should then be excluded from the list of results, the same as any org/env combination that doesn't exist.

Reply:
Ryan suggested organization to be one var instead of array, so I think this won't be necessary

Reply:
You may still have cases where the environments in a single organization have different target servers. If it is present in one environment, but not the other, then the entire script "breaks".

Since this is an open source library, you have to consider not only our typical environment configuration, but also the potential environments and use cases. We won't be able to cater for all of them, but still have to think more general.